any2dataset/downloader.py

```diff
@@ -9,11 +9,30 @@
 import hashlib
 import pyarrow as pa
 import traceback
+import subprocess
+import soundfile as sf
 
 import fsspec
 from .logger import CappedCounter
 from .logger import write_stats
 
+def download_file(row, timeout):
+    """Download a file with ffmpeg"""
+    key, url = row
+    file_stream = None
+    try:
+        cmd = f'ffmpeg -v error -i {url} -timeout {timeout}  -c copy -f s16le pipe:1'
+        p = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
That would open one process per thread. Seems unlikely to work

Reply:
So what is better? Use os.system instead subprocess or ffmpeg-python wrapper?

Reply:
That does the same thing.
Ffmpeg is another process

Spawning too many processes is too slow.

Doing ffmpeg during subsampling solves that issue by having N*M downloading threads and only N processes doing ffmpeg subsampling

Reply:
all right, thank you!